code/modules/mob/living/carbon/xenomorph/castes/lesser_drone.dm

```diff
@@ -83,6 +83,15 @@
 	weed_food_states = list("Lesser_Drone_1","Lesser_Drone_2","Lesser_Drone_3")
 	weed_food_states_flipped = list("Lesser_Drone_1","Lesser_Drone_2","Lesser_Drone_3")
 
+/mob/living/carbon/xenomorph/lesser_drone/Initialize(mapload, mob/living/carbon/xenomorph/old_xeno, hivenumber)
```

Comment:
This doesn't handle the situation of freed mobs or taking over an AFK xeno.

You probably want to use /mob/living/carbon/xenomorph/lesser_drone/Login instead or have this xeno register to COMSIG_GLOB_MOB_LOGIN to add the component. Or just always have the component on this xeno type and have it register to COMSIG_GLOB_MOB_LOGIN to start its say blocking & wait timer.

Reply:
all reviews seen to